### ISSUE TITLE
Suggestion to replace PinkySwear with a more updated implementation

### ISSUE THREAD

**Issue by paulocoghi** [pyrsmk/qwest#128]
PinkySwear is (probably) not maintained anymore.

The last update was made in July 30, 2014 and it uses Mocha 1.21.5, that uses Jade instead of Pug, extremely outdated Graceful-fs and a version of MiniMatch that has security vulnerabilities.

I really like [Zousan](https://github.com/bluejava/zousan) but there are also [many other implementations](https://github.com/promises-aplus/promises-spec/blob/master/implementations.md) and I don't want to be unfair.

I am not a super-advanced js programmer, and I feel that I don't have enough knowledge to make this replacement in Qwest myself.

But if it's interesting to you, I am available to learn and do this. :)

**Comment by pyrsmk** [comment]
Good idea, I should update that with Zousan. I need to analyze the current PinkySwear implementation first, because qwest is using some specific PinkySwear options (like Promise prototyping).

Since qwest need to be refactored, I plan this for the v5 release ;)
